src/current/advisories/a104309.md

```diff
@@ -0,0 +1,80 @@
+
+1. If the above conditions are satisfied, the follower attempts to push the transaction to advance its resolved timestamp and emit a checkpoint, but the follower does not find a transaction record. The follower operates as though the transaction was aborted, and allows its resolved timestamp to advance above the transaction's write timestamp, emitting a checkpoint even though the intents have not yet been emitted.
+
+1. When the intent resolution is finally replicated to the follower, it emits the write events with timestamps below the previous checkpoint.
+
+1. The changefeed processor detects these events below the previous checkpoint and discards them instead of emitting them.
+
+All CockroachDB versions prior to and including the following are impacted by this bug:
+
+- v22.2.17 and below
+- v23.1.0 - v23.1.14
+- V23.2.0
+
+Versions prior to 22.2 are no longer eligible for [maintenance support](https://www.cockroachlabs.com/docs/releases/release-support-policy), and this bug will not be fixed in those versions.
+
+## Statement
+
+This is resolved in CockroachDB by [PR #117612](https://github.com/cockroachdb/cockroach/pull/117612), which uses a barrier command to ensure that all historical and ongoing range writes have been applied to the local replica and emitted before the resolved timestamp is advanced and a checkpoint is emitted.
+
+The fix has been applied to maintenance releases of CockroachDB:
+
+- [v22.2.18](https://www.cockroachlabs.com/docs/releases/v22.2#v22-2-18)
+- [v23.1.15](https://www.cockroachlabs.com/docs/releases/v23.1#v23-1-15)
+- [v23.2.1](https://www.cockroachlabs.com/docs/releases/v23.2#v23-2-1)
+
+This public issue is tracked by [Issue #104309](https://github.com/cockroachdb/cockroach/issues/104309).
+
+## Mitigation
+
+Users are encouraged to upgrade to v22.2.18, v23.1.15, v23.2.1, or a later version that includes the fix.
+
+The log message `cdc ux violation: detected timestamp ... that is less or equal to the local frontier` indicates that a cluster has been affected by this bug. One message will be logged per omitted event. The message will include the [job ID](https://www.cockroachlabs.com/docs/v23.2/show-jobs#show-changefeed-jobs) of the affected changefeed (as `job=...`), as well as the MVCC timestamp of the transaction whose writes were omitted (`detected timestamp ...`). If the data is still present in a table, the following query may allow you to retrieve it:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT * FROM table WHERE crdb_internal_mvcc_timestamp = {timestamp}
```

Comment:
We'll need this qualifier:

Where `timestamp` must be converted to nanoseconds with the logical component after the decimal point. For example, given the logged timestamp 1707854079.38430848,2 multiply 1707854079 by 1e9 (1 billion) and add 38430848, then put 2 after the decimal point: 1707854079038430848.2

Reply:
Done, with some light tweaking.